Sources/Build/Buildable.swift

```diff
@@ -9,20 +9,43 @@
 */
 
 import PackageType
+import struct Utility.Path
 
 protocol Buildable {
     var targetName: String { get }
     var isTest: Bool { get }
 }
 
+extension CModule {
+    func workingDirectory(prefix: String) -> String {
+        return Path.join(prefix, "\(c99name).build")
+    }
+}
+
 extension Module: Buildable {
     var isTest: Bool {
         return self is TestModule
     }
 
-    var Xcc: [String] {
+    func XccFlagsForPrefix(prefix: String) -> [String] {
         return recursiveDependencies.flatMap { module -> [String] in
-            if let module = module as? CModule {
+            if let module = module as? ClangModule {
+                var moduleMap: String? = nil
+                
+                if module.moduleMapPath.isFile {
+                    moduleMap = module.moduleMapPath
+                }
+    
+                let genModuleMap = Path.join(module.workingDirectory(prefix), module.moduleMap)
+                if genModuleMap.isFile {
+                    moduleMap = genModuleMap
+                }
```

Comment:
I don't think we need to use absolute paths, although it may be there isn't a downside to doing so.

It does require not copying the headers if we want to drop the -fmodule-map-file arguments, but maybe there isn't a need to do that. Let's start out by trying to not copy the headers (which I agree is nice), and see how far we get.

The one thing that copying the headers does give us is that if the target author makes a mistake and refers to one of the headers from their source using, e.g., "../foo.h" then it would catch that error. That is the kind of thing that could break in the future if we decided to separate out the build of dependent targets, and then move their "products" (i.e. built binary and installable headers) around.

However, that probably isn't enough motivation by itself to copy things.